contracts/proxy/utils/Initializable.sol

```diff
@@ -55,14 +55,27 @@ pragma solidity ^0.8.19;
  */
 abstract contract Initializable {
     /**
-     * @dev Indicates that the contract has been initialized.
+     * @dev Storage of the initializable contract.
+     *
+     * It's implemented on a custom ERC-7201 namespace to reduce the risk of storage collisions
+     * when using with upgradeable contracts.
+     *
+     * @custom:storage-location erc7201:openzeppelin.storage.Initializable
      */
-    uint8 private _initialized;
+    struct InitializableStorage {
+        /**
+         * @dev Indicates that the contract has been initialized.
+         */
+        uint64 _initialized;
+        /**
+         * @dev Indicates that the contract is in the process of being initialized.
+         */
+        bool _initializing;
+    }
 
-    /**
-     * @dev Indicates that the contract is in the process of being initialized.
-     */
-    bool private _initializing;
+    // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.Initializable")) - 1))
+    bytes32 private constant _INITIALIZABLE_STORAGE =
+        0xf0c57e16840df040f15088dc2f81fe391c3923bec73e23a9662efc9c229c6a0e;
```

Comment:
We may want to do the computation differently.

Reply:
Ah, amazing catch! Note we haven't updated EIP-7201 to reflect the impact of Verkle Trees neither.

Are we ok with?:

```suggestion
    // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.Initializable")) - 256)) & ~bytes32(uint256(0xFF))
    bytes32 private constant _INITIALIZABLE_STORAGE =
        0x2cb0f341752e459a3615b447e07b189024fd0be4129b407c58ddd277d84b6800;
```

Reply:
Let's not change this until we change the EIP.